docs/tutorial.rst

```diff
@@ -0,0 +1,346 @@
+
+Importing need items
+--------------------
+
+Need items can also be imported from external sources, using the :ref:`needimport` directive,
+or generated from external services, using the :ref:`needservice` directive.
+
+Lets import some test cases, we add an additional tag to each, to make them easier to select later on:
+
+.. need-example:: Importing need items
+
+    .. needimport:: _static/tutorial_needs.json
+        :tags: tutorial,tutorial_tests
+        :collapse: true
+
+.. seealso::
+    
+    For full options see the reference sections for :ref:`needimport directive <needimport>` and :ref:`needservice directive <needservice>`.
+
+Modifying need items
+--------------------
+
+In the section above, we imported some test case needs, but they are currently not linked to any other need items.
+
+We can extend the imported need items using the :ref:`needextend directive <needextend>`,
+to add additional fields to them, such as links.
+
+The ``needextend`` directive expects a :ref:`filter <filter>` argument, which is used to select the need items to extend.
+Here we filter by the tag we set on the imported items above:
+
+.. need-example:: Extending need items
+
+    .. needextend:: "tutorial_tests" in tags
+        :+tutorial_tests: T_RADAR
+        :status: open
+
+    .. needextend:: T_001
+        :status: closed
+
+    .. needextend:: T_002
+        :status: in progress
+
+.. note:: 
+
+    The ``needextend`` does not have any visible output,
+    but it you look at the items, they will now have the additional link and status fields.
+
+.. seealso:: 
+    
+    For full options see the reference sections for :ref:`needextend directive <needextend>`.
+
+Summarising needs
+-----------------
+
+Now we have learnt about how to introduce need items into our project,
+it is natural to want to be able to summarise all or a sub-set of needs.
+
+There are three directives that can be used to do this, with different output formats:
+
+- :ref:`needlist <needlist>` - to display a list of need items
+- :ref:`needtable <needtable>` - to display a table of need items
+- :ref:`needflow <needflow>` - to display a flow diagram of need items
+
+All of these use a common :ref:`filter logic <filter>`, to select a sub-set of need items to display,
+either by simple options, or by using a more complex expression.
+
+In the following example we will display a list of all need items with the tag "tutorial",
+sorted by ID, and showing the status of each item:
+
+.. need-example:: Simple list
+
+    .. needlist::
+        :tags: tutorial
+        :sort_by: id
+        :show_status:
+
+Similarly, we can display the same items in a table format:
+
+.. need-example:: Simple table
+
+    .. needtable::
+        :tags: tutorial
+        :sort: id
+        :columns: id,type,title,status
+        :style: table
+
+The default style for the table is a simple HTML table, but we can also use the ``datatables`` style to add dynamic pagination, filtering and sorting,
```

Comment:
I think the default style is the datatables style, or did this change?

Reply:
nope you are correct, now fixed